MPU status-related tests failing

The following three test methods are failing:

- com.joyent.manta.client.multipart.EncryptedServerSideMultipartManagerIT#canAbortUpload

```
  java.lang.AssertionError: MPU is not in an aborted or aborting status. Actual status: UNKNOWN

    at org.testng.Assert.fail(Assert.java:94)
    at com.joyent.manta.client.multipart.EncryptedServerSideMultipartManagerIT.canAbortUpload(EncryptedServerSideMultipartManagerIT.java:109)
```
- com.joyent.manta.client.multipart.EncryptedServerSideMultipartManagerIT#canGetPart
```
  com.joyent.manta.exception.MantaMultipartException: Expected response field was missing or malformed
  Exception Context:
    [1:mantaSdkVersion=3.0.0-SNAPSHOT]
    [2:requestId=a82e57eb-57a1-11e7-9f7b-f132c1bbfa44]
    [3:request=HttpGet[version=<null>,uri=https://us-east.manta.samsungcloud.io/tomas.celaya/uploads/7/7e0fc5f6-7d24-e8cb-effa-d1611cfd8a50/state,config=<null>,aborted=false,cancellableRef=org.apache.http.impl.execchain.ConnectionHolder@65e21ce3,headergroup=[],params=org.apache.http.params.BasicHttpParams@2aff9dff]]
    [4:requestMethod=GET]
    [5:requestURL=https://us-east.manta.samsungcloud.io/tomas.celaya/uploads/7/7e0fc5f6-7d24-e8cb-effa-d1611cfd8a50/state]
    [6:requestHeaders=]
    [7:loadBalancerAddress=148.78.5.213]
    [8:response=HttpResponseProxy[original=HTTP/1.1 200 OK [Content-Type: application/json, Content-Length: 671, Content-MD5: GgT1nyPI59qTx5h/KzozwQ==, Date: Thu, 22 Jun 2017 23:22:28 GMT, Server: Manta, x-request-id: a82e57eb-57a1-11e7-9f7b-f132c1bbfa44, x-response-time: 5, x-server-name: c14c1fb8-8446-45fb-b557-df98395f7320, Connection: keep-alive] ResponseEntityProxy{[Content-Type: application/json,Content-Length: 671,Chunked: false]},connHolder=org.apache.http.impl.execchain.ConnectionHolder@65e21ce3]]
    [9:responseHeaders=Content-Type: application/json, Content-Length: 671, Content-MD5: GgT1nyPI59qTx5h/KzozwQ==, Date: Thu, 22 Jun 2017 23:22:28 GMT, Server: Manta, x-request-id: a82e57eb-57a1-11e7-9f7b-f132c1bbfa44, x-response-time: 5, x-server-name: c14c1fb8-8446-45fb-b557-df98395f7320, Connection: keep-alive]
    [10:responseStatusCode=200]
    [11:responseStatusReason=OK]
  ---------------------------------

    at com.joyent.manta.client.multipart.ServerSideMultipartManager.getPart(ServerSideMultipartManager.java:378)
    at com.joyent.manta.client.multipart.ServerSideMultipartManager.getPart(ServerSideMultipartManager.java:67)
    at com.joyent.manta.client.multipart.EncryptedMultipartManager.getPart(EncryptedMultipartManager.java:326)
    at com.joyent.manta.client.multipart.EncryptedServerSideMultipartManagerIT.canGetPart(EncryptedServerSideMultipartManagerIT.java:178)

  Caused by: java.lang.NullPointerException: Unable to read object path from response
    at org.apache.commons.lang3.Validate.notNull(Validate.java:222)
    at com.joyent.manta.client.multipart.ServerSideMultipartManager.getPart(ServerSideMultipartManager.java:368)
    ... 26 more
```
- com.joyent.manta.client.multipart.EncryptedServerSideMultipartManagerIT#canGetStatus
```
java.lang.AssertionError: MPU com.joyent.manta.client.multipart.ServerSideMultipartUpload@151bf776[id=3bdbbf90-5096-e68a-fe34-d81915faf78f,path=/tomas.celaya/stor/java-manta-integration-tests/16c93e5e-d580-4822-a127-426d969f43d85cd108d2-8e2a-481f-820d-d735bcb7e636,partsDirectory=/tomas.celaya/uploads/3/3bdbbf90-5096-e68a-fe34-d81915faf78f]in invalid status after complete invocation. Actual status: UNKNOWN

    at org.testng.Assert.fail(Assert.java:94)
    at com.joyent.manta.client.multipart.EncryptedServerSideMultipartManagerIT.canGetStatus(EncryptedServerSideMultipartManagerIT.java:162)
```

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -456,9 +456,9 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
                     case "created":
                         return MantaMultipartStatus.CREATED;
                     case "finalizing":
-                        return extractMultipartStatusResult(get, response, objectNode, state);
+                        return extractMultipartStatusResult(get, response, objectNode);
                     case "done":
-                        return extractMultipartStatusResult(get, response, objectNode, state);
+                        return extractMultipartStatusResult(get, response, objectNode);
                     default:
                         final String stateMsg = "Expected response field was missing or malformed: state: " + state;
                         final MantaMultipartException e = new MantaMultipartException(stateMsg);
@@ -479,7 +479,18 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
         }
     }
 
-    private MantaMultipartStatus extractMultipartStatusResult(HttpGet get, CloseableHttpResponse response, ObjectNode objectNode, String state) {
+    /**
+     * Extract the "result" field from the get-mpu payload.
+     *
+     * @param get           The request for MPU status. Used for exception annotation.
+     * @param response      The response received. Also used for exception annotation.
+     * @param objectNode    The response JSON object.
+     * @return MantaMultipartStatus extracted
+     */
+    private MantaMultipartStatus extractMultipartStatusResult(
+            final HttpGet get,
+            final CloseableHttpResponse response,
+            final ObjectNode objectNode) {
         JsonNode resultNode = objectNode.get("result");
         Validate.notNull(resultNode, "Unable to get result from response");
         String result = resultNode.textValue();

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -49,6 +49,7 @@ import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 import java.util.Collection;
+import java.util.Locale;
 import java.util.Set;
 import java.util.UUID;
 import java.util.stream.Collectors;
@@ -451,25 +452,20 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
                 String state = stateNode.textValue();
                 Validate.notBlank(state, "State field was blank in response");
 
-                if (state.equalsIgnoreCase("created")) {
-                    return MantaMultipartStatus.CREATED;
+                switch (state) {
+                    case "created":
+                        return MantaMultipartStatus.CREATED;
+                    case "finalizing":
+                        return extractMultipartStatusResult(get, response, objectNode, state);
+                    case "done":
+                        return extractMultipartStatusResult(get, response, objectNode, state);
+                        // return MantaMultipartStatus.COMPLETED;
+                    default:
+                        final String stateMsg = "Expected response field was missing or malformed: state: " + state;
+                        final MantaMultipartException e = new MantaMultipartException(stateMsg);
+                        annotateException(e, get, response, null, null);
+                        throw e;
                 }
-
-                if (state.equalsIgnoreCase("finalizing")) {
-                    JsonNode typeNode = objectNode.get("type");
-                    Validate.notNull(typeNode, "Unable to get type from response");
-                    String type = typeNode.textValue();
-                    Validate.notBlank(type, "Type field was blank in response");
-
-                    if (type.equalsIgnoreCase("commit")) {
-                        return MantaMultipartStatus.COMMITTING;
-                    }
-                    if (type.equalsIgnoreCase("abort")) {
-                        return MantaMultipartStatus.ABORTING;
-                    }
-                }
-
-                return MantaMultipartStatus.UNKNOWN;
             } catch (JsonParseException e) {
                 String msg = "Response body was not JSON";
                 MantaMultipartException me = new MantaMultipartException(msg, e);
@@ -481,6 +477,29 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
                 annotateException(me, get, response, null, null);
                 throw me;
             }
+        }
+    }
+
+    private MantaMultipartStatus extractMultipartStatusResult(HttpGet get, CloseableHttpResponse response, ObjectNode objectNode, String state) {
+        JsonNode detailNode = objectNode.get("result");
+        Validate.notNull(detailNode, "Unable to get result from response");
+        String detail = detailNode.textValue();
+        Validate.notBlank(detail, "Result field was blank in response");
+
+        switch (detail.toLowerCase(Locale.ROOT)) {
+            case "aborting":
+                return MantaMultipartStatus.ABORTING;
+            case "aborted":
+                return MantaMultipartStatus.ABORTED;
+            case "committing":
+                return MantaMultipartStatus.COMMITTING;
+            case "committed":
+                return MantaMultipartStatus.COMMITTING;
+            default:
+                final String stateTypeMsg = "Result response field was missing or malformed: " + detail;
+                final MantaMultipartException e = new MantaMultipartException(stateTypeMsg);
+                annotateException(e, get, response, null, null);
+                throw e;
         }
     }
 

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -452,14 +452,13 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
                 String state = stateNode.textValue();
                 Validate.notBlank(state, "State field was blank in response");
 
-                switch (state) {
+                switch (state.toLowerCase(Locale.ROOT)) {
                     case "created":
                         return MantaMultipartStatus.CREATED;
                     case "finalizing":
                         return extractMultipartStatusResult(get, response, objectNode, state);
                     case "done":
                         return extractMultipartStatusResult(get, response, objectNode, state);
-                        // return MantaMultipartStatus.COMPLETED;
                     default:
                         final String stateMsg = "Expected response field was missing or malformed: state: " + state;
                         final MantaMultipartException e = new MantaMultipartException(stateMsg);

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -49,7 +49,6 @@ import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 import java.util.Collection;
-import java.util.Locale;
 import java.util.Set;
 import java.util.UUID;
 import java.util.stream.Collectors;

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -480,12 +480,12 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
     }
 
     private MantaMultipartStatus extractMultipartStatusResult(HttpGet get, CloseableHttpResponse response, ObjectNode objectNode, String state) {
-        JsonNode detailNode = objectNode.get("result");
-        Validate.notNull(detailNode, "Unable to get result from response");
-        String detail = detailNode.textValue();
-        Validate.notBlank(detail, "Result field was blank in response");
+        JsonNode resultNode = objectNode.get("result");
+        Validate.notNull(resultNode, "Unable to get result from response");
+        String result = resultNode.textValue();
+        Validate.notBlank(result, "Result field was blank in response");
 
-        switch (detail.toLowerCase(Locale.ROOT)) {
+        switch (result.toLowerCase(Locale.ROOT)) {
             case "aborting":
                 return MantaMultipartStatus.ABORTING;
             case "aborted":
@@ -495,7 +495,7 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
             case "committed":
                 return MantaMultipartStatus.COMMITTING;
             default:
-                final String stateTypeMsg = "Result response field was missing or malformed: " + detail;
+                final String stateTypeMsg = "Expected response field was missing or malformed: result: " + result;
                 final MantaMultipartException e = new MantaMultipartException(stateTypeMsg);
                 annotateException(e, get, response, null, null);
                 throw e;

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -456,14 +456,11 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
                     case "created":
                         return MantaMultipartStatus.CREATED;
                     case "finalizing":
-                        return extractMultipartStatusResult(get, response, objectNode);
+                        return extractMultipartStatusResult(objectNode);
                     case "done":
-                        return extractMultipartStatusResult(get, response, objectNode);
+                        return extractMultipartStatusResult(objectNode);
                     default:
-                        final String stateMsg = "Expected response field was missing or malformed: state: " + state;
-                        final MantaMultipartException e = new MantaMultipartException(stateMsg);
-                        annotateException(e, get, response, null, null);
-                        throw e;
+                        return MantaMultipartStatus.UNKNOWN;
                 }
             } catch (JsonParseException e) {
                 String msg = "Response body was not JSON";
@@ -487,10 +484,7 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
      * @param objectNode    The response JSON object.
      * @return MantaMultipartStatus extracted
      */
-    private MantaMultipartStatus extractMultipartStatusResult(
-            final HttpGet get,
-            final CloseableHttpResponse response,
-            final ObjectNode objectNode) {
+    private MantaMultipartStatus extractMultipartStatusResult(final ObjectNode objectNode) {
         JsonNode resultNode = objectNode.get("result");
         Validate.notNull(resultNode, "Unable to get result from response");
         String result = resultNode.textValue();
@@ -506,10 +500,7 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
             case "committed":
                 return MantaMultipartStatus.COMMITTING;
             default:
-                final String stateTypeMsg = "Expected response field was missing or malformed: result: " + result;
-                final MantaMultipartException e = new MantaMultipartException(stateTypeMsg);
-                annotateException(e, get, response, null, null);
-                throw e;
+                return MantaMultipartStatus.UNKNOWN;
         }
     }
 

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -365,7 +365,7 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
             try (InputStream in = response.getEntity().getContent()) {
                 ObjectNode objectNode = MantaObjectMapper.INSTANCE.readValue(in, ObjectNode.class);
 
-                JsonNode objectPathNode = objectNode.get("objectPath");
+                JsonNode objectPathNode = objectNode.get("targetObject");
                 Validate.notNull(objectPathNode, "Unable to read object path from response");
                 objectPath = objectPathNode.textValue();
                 Validate.notBlank(objectPath, "Object path field was blank in response");

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -479,8 +479,6 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
     /**
      * Extract the "result" field from the get-mpu payload.
      *
-     * @param get           The request for MPU status. Used for exception annotation.
-     * @param response      The response received. Also used for exception annotation.
      * @param objectNode    The response JSON object.
      * @return MantaMultipartStatus extracted
      */

--- a/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
+++ b/java-manta-client/src/main/java/com/joyent/manta/client/multipart/ServerSideMultipartManager.java
@@ -452,16 +452,17 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
                 String state = stateNode.textValue();
                 Validate.notBlank(state, "State field was blank in response");
 
-                switch (state.toLowerCase(Locale.ROOT)) {
-                    case "created":
-                        return MantaMultipartStatus.CREATED;
-                    case "finalizing":
-                        return extractMultipartStatusResult(objectNode);
-                    case "done":
-                        return extractMultipartStatusResult(objectNode);
-                    default:
-                        return MantaMultipartStatus.UNKNOWN;
+                if (state.equalsIgnoreCase("created")) {
+                    return MantaMultipartStatus.CREATED;
                 }
+                if (state.equalsIgnoreCase("finalizing")) {
+                    return extractMultipartStatusResult(objectNode);
+                }
+                if (state.equalsIgnoreCase("done")) {
+                    return extractMultipartStatusResult(objectNode);
+                }
+
+                return MantaMultipartStatus.UNKNOWN;
             } catch (JsonParseException e) {
                 String msg = "Response body was not JSON";
                 MantaMultipartException me = new MantaMultipartException(msg, e);
@@ -488,18 +489,20 @@ public class ServerSideMultipartManager extends AbstractMultipartManager
         String result = resultNode.textValue();
         Validate.notBlank(result, "Result field was blank in response");
 
-        switch (result.toLowerCase(Locale.ROOT)) {
-            case "aborting":
-                return MantaMultipartStatus.ABORTING;
-            case "aborted":
-                return MantaMultipartStatus.ABORTED;
-            case "committing":
-                return MantaMultipartStatus.COMMITTING;
-            case "committed":
-                return MantaMultipartStatus.COMMITTING;
-            default:
-                return MantaMultipartStatus.UNKNOWN;
+        if (result.equalsIgnoreCase("aborting")) {
+            return MantaMultipartStatus.ABORTING;
         }
+        if (result.equalsIgnoreCase("aborted")) {
+            return MantaMultipartStatus.ABORTED;
+        }
+        if (result.equalsIgnoreCase("committing")) {
+            return MantaMultipartStatus.COMMITTING;
+        }
+        if (result.equalsIgnoreCase("committed")) {
+            return MantaMultipartStatus.COMMITTING;
+        }
+
+        return MantaMultipartStatus.UNKNOWN;
     }
 
     @Override

